File: frontend/types/notification.ts
export interface Notification {
  id: number;
  user_id: string;
  task_id: number | null;
  type: 'due_soon' | 'overdue';
  message: string;
  is_read: boolean;
  sent_email: boolean;
  created_at: string;
}

export interface NotificationListResponse {
  notifications: Notification[];
  unread_count: number;
}

File: backend/src/api/schemas/notification.py
"""Pydantic schemas for Notification operations."""

from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class NotificationResponse(BaseModel):
    """Schema for a single notification."""

    id: int
    user_id: str
    task_id: Optional[int]
    type: str
    message: str
    is_read: bool
    sent_email: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class NotificationListResponse(BaseModel):
    """Schema for notification list with unread count."""

    notifications: list[NotificationResponse]
    unread_count: int
